ni_measurementlink_service/_internal/parameter/serialization_strategy.py

```diff
@@ -195,17 +212,22 @@ def get_encoder(type: type_pb2.Field.Kind.ValueType, repeated: bool) -> PartialE
     return scalar
 
 
-def get_decoder(type: type_pb2.Field.Kind.ValueType, repeated: bool) -> PartialDecoderConstructor:
-    """Get the appropriate partial decoder constructor for the specified type.
-
-    A scalar or vector constructor is returned based on the 'repeated' parameter.
-    """
-    if type not in _FIELD_TYPE_TO_DECODER_MAPPING:
+def get_decoder(
+    type: type_pb2.Field.Kind.ValueType, repeated: bool, message_type: str = ""
+) -> PartialDecoderConstructor:
+    """Get the appropriate partial decoder constructor for the specified type."""
+    decoder_mapping = _FIELD_TYPE_TO_DECODER_MAPPING.get(type)
+    if decoder_mapping is not None:
+        scalar_decoder, array_decoder = decoder_mapping
+        return array_decoder if repeated else scalar_decoder
+    elif message_type == xydata_pb2.DoubleXYData.DESCRIPTOR.full_name:
+        if type != type_pb2.Field.Kind.TYPE_MESSAGE:
+            raise ValueError(f"Message type must have a TYPE_MESSAGE kind '{message_type}'")
+        if repeated:
+            raise ValueError(f"Repeated message types are not supported '{message_type}'")
+        return XYDataDecoder
```

Comment:
Let's structure this so we can add more message types without duplicating the error checks. 

```suggestion
    elif type == type_pb2.Field.Kind.TYPE_MESSAGE:
        if repeated:
            raise ValueError(f"Repeated message types are not supported '{message_type}'")
        if message_type == xydata_pb2.DoubleXYData.DESCRIPTOR.full_name:
            return XYDataDecoder
        else:
            raise ValueError(f"Unknown message type '{message_type}'")
```

Reply:
Even better... (didn't I already suggest this?)
```suggestion
    elif type == type_pb2.Field.Kind.TYPE_MESSAGE:
        if repeated:
            raise ValueError(f"Repeated message types are not supported '{message_type}'")
        decoder = _MESSAGE_TYPE_TO_DECODER.get(message_type)
        if decoder is None:
            raise ValueError(f"Unknown message type '{message_type}'")
        return decoder
```

Reply:
Yeah, you did suggest a second dictionary for _MESSAGE_TYPE_TO_DECODER. I was going to defer this until we add more message types, but I'll go ahead and add it.